src/main/flight/position.c

```diff
@@ -47,15 +47,14 @@
 #include "pg/pg.h"
 #include "pg/pg_ids.h"
 
-static float estimatedAltitudeCm = 0;
-static float estimatedAltitudeDerivative = 0;
-
-#ifdef USE_VARIO
-static int16_t estimatedVario = 0;                   // in cm/s
-#endif
-
+static float displayAltitudeCm = 0.0f;
+static float zeroedAltitudeCm = 0.0f;
```

Comment:
Why not keep it `estimated` instead of `zeroed` for all changes made.

Reply:
There are two different altitude values that we need.

For the OSD, to provide feedback to the pilot, we show, while disarmed:
- the Above Sea Level (or absolute) GPS value, if GPS is available
- the Baro altitude value after 'calibration' at power-up, if no GPS

While armed, the OSD shows the relative altitude of the quad to home, which is the value that we use to control altitude, and is a mix of GPS and Baro altitude data.

It would be nice to have only one value, which switches between the two.

However the value to use for flight control while armed must be filtered.  The filter will carry a 'memory' of recent values.  If the input to the filtered, zeroed value includes previous Above Sea Level values, then there will be a transient spike in the altitude value when we arm.

This transient spike was setting Max Altitude inappropriately high for pilots operating at high altitude.  Hence the input to the filtered value must be forced to zero while disarmed.  

The rewrite separates out the 'zeroed while disarmed' altitude value which enters the filter from the 'un-filtered above sea level' value.